File: src/main/java/com/procurement/notice/model/ocds/RequirementReference.java
package com.procurement.notice.model.ocds;

import com.fasterxml.jackson.annotation.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "id",
        "title"
})
public class RequirementReference {
    @JsonProperty("id")
    @JsonPropertyDescription("The id of the requirement which the response is applicable to")
    private String id;

    @JsonProperty("title")
    @JsonPropertyDescription("The title of the requirement which the response is applicable to")
    private final String title;

    @JsonCreator
    public RequirementReference(@JsonProperty("id") final String id,
                                @JsonProperty("title") final String title) {
        this.id = id;
        this.title = title;
    }
}

File: src/main/kotlin/com/procurement/notice/model/ocds/RequirementReference.kt
package com.procurement.notice.model.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude

data class RequirementReference @JsonCreator constructor(

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val id: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val title: String?
)
